scenario/context.py

```diff
@@ -450,6 +451,7 @@ def __init__(
         self.juju_log: List["JujuLogLine"] = []
         self.app_status_history: List["_EntityStatus"] = []
         self.unit_status_history: List["_EntityStatus"] = []
+        self.exec_history = collections.defaultdict(list)
```

Comment:
Why is this a defaultdict? To me it seems like non-existent containers be a KeyError?

Reply:
Hmm, fair point. A `defaultdict` was the simple solution because the `Context` object has no idea what containers will exist. But I should just change the appending to handle the list not existing.